packages/theme-product-form/__tests__/theme-product-form.test.js

```diff
@@ -0,0 +1,311 @@
+    const submitEvent = new Event('submit');
+    const options = {
+      onFormSubmit: jest.fn()
+    };
+
+    const productForm = new ProductForm(element, productJSON, options);
+
+    productForm.element.dispatchEvent(submitEvent);
+
+    expect(options.onFormSubmit).toHaveBeenCalledWith(submitEvent);
+    expectFormEventDataset(submitEvent, {});
+  });
+});
+
+describe('ProductForm.destroy()', () => {
+  test('removes all event listeners that were assigned by ProductForm()', () => {
+    const element = document.getElementById('form');
+    const colorSelect = document.querySelector('[name="options[Color]"]');
+    const quantityElement = document.querySelector('[name="quantity"]');
+    const propertyElement = document.querySelector('[name^="properties"]');
+
+    const changeEvent = new Event('change');
+    const submitEvent = new Event('submit');
+
+    const config = {
+      onOptionChange: jest.fn(),
+      onQuantityChange: jest.fn(),
+      onPropertyChange: jest.fn(),
+      onFormSubmit: jest.fn()
+    };
+
+    const productForm = new ProductForm(element, productJSON, config);
+
+    productForm.destroy();
+
+    colorSelect.dispatchEvent(changeEvent);
+    quantityElement.dispatchEvent(changeEvent);
+    propertyElement.dispatchEvent(changeEvent);
+    productForm.element.dispatchEvent(submitEvent);
+
+    expect(config.onOptionChange).not.toHaveBeenCalled();
+    expect(config.onQuantityChange).not.toHaveBeenCalled();
+    expect(config.onPropertyChange).not.toHaveBeenCalled();
+    expect(config.onFormSubmit).not.toHaveBeenCalled();
+  });
+});
+
+describe('ProductForm.options()', () => {
+  test('returns an ordered array of option values', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.options()).toMatchObject(defaultOptions);
+  });
+});
+
+describe('ProductForm.variant()', () => {
+  test('returns the current variant of the form', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.variant()).toMatchObject(defaultVariant);
+  });
+});
+
+describe('ProductForm.properties()', () => {
+  test('returns a collection of objects containing the name and value of properties', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.properties()).toMatchObject(defaultProperties);
+  });
+});
+
+describe('ProductForm.quantity()', () => {
+  test('returns the current quantity specified in the form', () => {
+    const element = document.getElementById('form');
+    const productForm = new ProductForm(element, productJSON);
+
+    expect(productForm.quantity()).toBe(defaultQuantity);
+  });
+
+  test('returns 1 if no quantity input exists in the form', () => {
+    document.body.innerHTML = `<form id="form"></form>`;
```

Comment:
Do you need back ticks here or could you use single quotes?

Reply:
Does it really matter?